projects/storefrontlib/src/layout/theme/site-theme.service.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ComponentRef,
+  Injectable,
+  Renderer2,
+  RendererFactory2,
+} from '@angular/core';
+import { SiteContextConfig, THEME_CONTEXT_ID } from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class SiteThemeService {
+  protected rootComponent: ComponentRef<any>;
+  protected renderer: Renderer2;
+  protected existingTheme: string;
+
+  constructor(
+    protected config: SiteContextConfig,
+    protected rendererFactory: RendererFactory2
+  ) {}
+
+  init(rootComponent: ComponentRef<any>): void {
+    this.renderer = this.rendererFactory.createRenderer(null, null);
+    this.rootComponent = rootComponent;
+    this.setTheme(this.config.context[THEME_CONTEXT_ID]?.[0]);
+  }
+
+  setTheme(theme: string): void {
+    if (theme) {
+      const element = this.rootComponent.location.nativeElement;
+      // remove the old theme
+      if (this.existingTheme) {
```

Comment:
nitpick: I'd still argue if this is needed, as the `removeClass` method  will not throw an error in case the class is not available.

Reply:
actually, I've missed the intent before; whenever there's no previous set theme, it's indeed a bit weird to call directly `removeClass`. My bad, feel free to revert this line.